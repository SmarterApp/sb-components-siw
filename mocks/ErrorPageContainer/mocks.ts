import { pageType, ErrorPageContainerProps } from "@src/index";

export const serverErrMock: ErrorPageContainerProps = {
  errorCode: pageType.ServerError,
  errorMsg: "you did a bad thing that caused errors",
  description: "stop doing that thing that caused errors"
};

export const notFoundErrMock: ErrorPageContainerProps = {
  errorCode: pageType.NotFound,
  errorMsg: "This item cannot be found",
  description: "Please try another search"
};
