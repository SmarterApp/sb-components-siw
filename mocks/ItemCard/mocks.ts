import { ItemCardModel, ItemTableContainerProps } from "@src/index";
import { ItemCardProps } from "@src/ItemCard/ItemCard";

export const completeItemCard: ItemCardModel = {
  selected: false,
  bankKey: 187,
  itemKey: 3434,
  title: "ELA Grade 4",
  grade: 4,
  gradeLabel: "Grade 4",
  subjectCode: "ELA",
  subjectLabel: "ELA",
  claimCode: "ELA3",
  claimLabel: "Claim",
  targetHash: 33434,
  targetShortName: "target short name",
  targetId: "dfas",
  targetDescription: "target description right here",
  interactionTypeCode: "GI",
  interactionTypeLabel: "Item Type",
  isPerformanceItem: false,
  brailleOnlyItem: false,
  domain: "domain",
  depthOfKnowledge: "2",
  commonCoreStandardId: "AB-123",
  ccssDescription: "this is the common core description",
  calculator: false,
  itemDifficulty: "Hard",
  availableBrailleTypes: [
    {
      label: "EBAE-Contracted",
      selectionCode: "EABE",
      disabled: false,
      hidden: false
    },
    {
      label: "BAET-Contracted",
      selectionCode: "BAET",
      disabled: false,
      hidden: false
    }
  ]

  // selected: false,
  // bankKey: 187,
  // itemKey: 3434,
  // interactionTypeCode: "GI",
  // interactionTypeLabel: "Item Type",
  // targetShortName: "Short Name",
  // title: "ELA Grade 4",
  // grade: 4,
  // gradeLabel: "Grade 4",
  // subjectCode: "ELA",
  // subjectLabel: "ELA",
  // claimCode: "ELA3",
  // claimLabel: "Claim",
  // targetHash: 33434,
  // targetDescription: "target description right hereeeee",
  // isPerformanceItem: false,
  // brailleOnlyItem: false,
  // targetId: "dfas",
  // depthOfKnowledge: "2",
  // commonCoreStandardId: "AB-123",
  // ccssDescription: "this is the common core description",
  // domain: "domain"
};

export const itemCardList: ItemCardModel[] = [
  {
    bankKey: 187,
    itemKey: 3206,
    stimulusKey: 56900,
    title: "Math Grade 6 Claim 1",
    grade: 8,
    gradeLabel: "Grade 6",
    subjectCode: "MATH",
    subjectLabel: "Math",
    testName: "Test-Name-1",
    testOrder: 1,
    claimCode: "MATH1",
    claimLabel: "1.Concepts and Procedures and Logical",
    targetHash: 100,
    targetId: "A",
    targetDescription: "Target A description with extra text for testing",
    targetShortName: "",
    interactionTypeCode: "EQ",
    interactionTypeLabel: "Equation Multiple Choice choice",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    calculator: true,
    commonCoreStandardId: "6.M.1a",
    itemDifficulty: "Moderate",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  },
  {
    bankKey: 187,
    itemKey: 3163,
    stimulusKey: 58679,
    title: "ELA/ literacy Grade 3 Claim 1",
    grade: 1,
    gradeLabel: "Grade 3",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    testName: "Test-Name-1",
    testOrder: 2,
    claimCode: "ELA1",
    claimLabel: "1.Reading",
    targetHash: 1039,
    targetId: "B",
    targetDescription: "Target B description",
    targetShortName: "Key Details",
    interactionTypeCode: "MC",
    interactionTypeLabel: "Multiple Choice",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    calculator: false,
    commonCoreStandardId: "3.R.2a",
    itemDifficulty: "Easy",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  },
  {
    bankKey: 187,
    itemKey: 3615,
    stimulusKey: 23461,
    title: "Math Grade 6 Claim 4",
    grade: 8,
    gradeLabel: "Grade 6",
    subjectCode: "MATH",
    subjectLabel: "Math",
    testName: "Test-Name-1",
    testOrder: 3,
    claimCode: "MATH4",
    claimLabel: "3.Modeling/Data Analysis",
    targetHash: 0,
    targetId: "C",
    targetDescription: "target C description",
    targetShortName: "",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    commonCoreStandardId: "6.M.2b",
    itemDifficulty: "",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  },
  {
    bankKey: 187,
    itemKey: 2928,
    stimulusKey: 87679,
    title: "ELA/ literacy Grade 4 Claim 4",
    grade: 2,
    gradeLabel: "Grade 4",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    testName: "",
    testOrder: 2,
    claimCode: "ELA4",
    claimLabel: "4.Research/Inquiry",
    targetHash: 2832,
    targetId: "D",
    targetDescription: "target D description",
    targetShortName: "Evaluate Information/ sources",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false,
    commonCoreStandardId: "4.E.4a",
    itemDifficulty: "Hard",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: true,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  }
];

export const totalItemsCardList: ItemCardModel[] = itemCardList;

export const sortableItemCards: ItemCardModel[] = [
  {
    bankKey: 187,
    itemKey: 1,
    title: "alpha",
    grade: 5,
    gradeLabel: "Grade 5",
    subjectCode: "MATH",
    subjectLabel: "Math",
    claimCode: "MATH1",
    claimLabel: "Alpha",
    targetHash: 1,
    targetId: "A",
    targetDescription: "target A description",
    targetShortName: "",
    interactionTypeCode: "EQ",
    interactionTypeLabel: "Equation",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    commonCoreStandardId: "",
    itemDifficulty: "Easy",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  },
  {
    bankKey: 187,
    itemKey: 2,
    title: "bravo",
    grade: 4,
    gradeLabel: "Grade 4",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    claimCode: "ELA1",
    claimLabel: "Bravo",
    targetHash: 2,
    targetId: "B",
    targetDescription: "target B description",
    targetShortName: "",
    interactionTypeCode: "MC",
    interactionTypeLabel: "Multiple Choice",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    commonCoreStandardId: "",
    itemDifficulty: "Moderate",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  },
  {
    bankKey: 187,
    itemKey: 3,
    title: "charlie",
    grade: 3,
    gradeLabel: "Grade 3",
    subjectCode: "MATH",
    subjectLabel: "Math",
    claimCode: "MATH4",
    claimLabel: "Charlie",
    targetHash: 3,
    targetId: "C",
    targetDescription: "target C description",
    targetShortName: "",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false,
    commonCoreStandardId: "",
    itemDifficulty: "Hard",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  },
  {
    bankKey: 187,
    itemKey: 4,
    title: "delta",
    grade: 2,
    gradeLabel: "Grade 2",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    claimCode: "ELA4",
    claimLabel: "Delta",
    targetHash: 4,
    targetId: "D",
    targetDescription: "target D description",
    targetShortName: "",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false,
    commonCoreStandardId: "",
    itemDifficulty: "Hard",
    availableBrailleTypes: [
      {
        label: "TDS BT ECN",
        selectionCode: "TDS_BT_ECN",
        disabled: false,
        hidden: false
      },
      {
        label: "DS BT UCB Label",
        selectionCode: "TDS_BT_UCB",
        disabled: false,
        hidden: false
      },
      {
        label: "TDS BT ECL label",
        selectionCode: "TDS_BT_ECL",
        disabled: false,
        hidden: false
      },
      {
        label: "TTDS BT EXN Label",
        selectionCode: "TTDS_BT_EXN",
        disabled: false,
        hidden: false
      }
    ]
  }
];

export const itemCardProps: ItemCardProps = {
  rowData: completeItemCard,
  onRowSelect: () => {
    return;
  },
  isInterimSite: true,
  getSelectedItemCount: () => {
    return 0;
  },
  showErrorModalOnPrintItemsCountExceeded: () => {
    return;
  },
  isPrintLimitEnabled: true,
  associatedItems: [],
  countNumberOfItemsAfterSelection: () => {
    return 0;
  },
  testCodeToLabelMap: { "Test-Name-1": "Test 1", "Test-Name-2": "Test 2" }
};

//PerformanceTaskAssociatedItems mockup data
export const performanceTaskAssociatedItems: any = {
  3206: [3163, 3206, 3615],
  3163: [3163, 3206, 3615],
  3615: [3163, 3206, 3615]
};

//PerformanceTaskAssociatedItems details data mockup
export const PTassociatedItems: any = {
  3206: [
    [
      {
        bankKey: 187,
        itemKey: 3163,
        title: "ELA/ literacy Grade 3 Claim 1",
        grade: 1,
        gradeLabel: "Grade 3",
        subjectCode: "ELA",
        subjectLabel: "ELA/literacy",
        claimCode: "ELA1",
        claimLabel: "Reading",
        targetHash: 1039,
        targetId: "B",
        targetDescription: "Target B description",
        targetShortName: "Key Details",
        interactionTypeCode: "MC",
        interactionTypeLabel: "Multiple Choice",
        isPerformanceItem: true,
        brailleOnlyItem: false,
        calculator: false
      }
    ],
    [
      {
        bankKey: 187,
        itemKey: 3206,
        title: "Math Grade 6 Claim 1",
        grade: 8,
        gradeLabel: "Grade 6",
        subjectCode: "MATH",
        subjectLabel: "Math",
        claimCode: "MATH1",
        claimLabel: "Concepts and Procedures and Logical",
        targetHash: 100,
        targetId: "A",
        targetDescription:
          "Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia",
        targetShortName: "",
        interactionTypeCode: "EQ",
        interactionTypeLabel: "Equation Multiple Choice choice",
        isPerformanceItem: true,
        brailleOnlyItem: false,
        calculator: true
      }
    ],
    [
      {
        bankKey: 187,
        itemKey: 3615,
        title: "Math Grade 6 Claim 4",
        grade: 8,
        gradeLabel: "Grade 6",
        subjectCode: "MATH",
        subjectLabel: "Math",
        claimCode: "MATH4",
        claimLabel: "Modeling/Data Analysis",
        targetHash: 0,
        targetId: "C",
        targetDescription: "target C description",
        targetShortName: "",
        interactionTypeCode: "MS",
        interactionTypeLabel: "Multi Select",
        isPerformanceItem: true,
        brailleOnlyItem: false
      }
    ]
  ],
  3163: [
    [
      {
        bankKey: 187,
        itemKey: 3163,
        title: "ELA/ literacy Grade 3 Claim 1",
        grade: 1,
        gradeLabel: "Grade 3",
        subjectCode: "ELA",
        subjectLabel: "ELA/literacy",
        claimCode: "ELA1",
        claimLabel: "Reading",
        targetHash: 1039,
        targetId: "B",
        targetDescription: "Target B description",
        targetShortName: "Key Details",
        interactionTypeCode: "MC",
        interactionTypeLabel: "Multiple Choice",
        isPerformanceItem: true,
        brailleOnlyItem: false,
        calculator: false
      }
    ],
    [
      {
        bankKey: 187,
        itemKey: 3206,
        title: "Math Grade 6 Claim 1",
        grade: 8,
        gradeLabel: "Grade 6",
        subjectCode: "MATH",
        subjectLabel: "Math",
        claimCode: "MATH1",
        claimLabel: "Concepts and Procedures and Logical",
        targetHash: 100,
        targetId: "A",
        targetDescription:
          "Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia",
        targetShortName: "",
        interactionTypeCode: "EQ",
        interactionTypeLabel: "Equation Multiple Choice choice",
        isPerformanceItem: true,
        brailleOnlyItem: false,
        calculator: true
      }
    ],
    [
      {
        bankKey: 187,
        itemKey: 3615,
        title: "Math Grade 6 Claim 4",
        grade: 8,
        gradeLabel: "Grade 6",
        subjectCode: "MATH",
        subjectLabel: "Math",
        claimCode: "MATH4",
        claimLabel: "Modeling/Data Analysis",
        targetHash: 0,
        targetId: "C",
        targetDescription: "target C description",
        targetShortName: "",
        interactionTypeCode: "MS",
        interactionTypeLabel: "Multi Select",
        isPerformanceItem: true,
        brailleOnlyItem: false
      }
    ]
  ],
  3615: [
    [
      {
        bankKey: 187,
        itemKey: 3163,
        title: "ELA/ literacy Grade 3 Claim 1",
        grade: 1,
        gradeLabel: "Grade 3",
        subjectCode: "ELA",
        subjectLabel: "ELA/literacy",
        claimCode: "ELA1",
        claimLabel: "Reading",
        targetHash: 1039,
        targetId: "B",
        targetDescription: "Target B description",
        targetShortName: "Key Details",
        interactionTypeCode: "MC",
        interactionTypeLabel: "Multiple Choice",
        isPerformanceItem: true,
        brailleOnlyItem: false,
        calculator: false
      }
    ],
    [
      {
        bankKey: 187,
        itemKey: 3206,
        title: "Math Grade 6 Claim 1",
        grade: 8,
        gradeLabel: "Grade 6",
        subjectCode: "MATH",
        subjectLabel: "Math",
        claimCode: "MATH1",
        claimLabel: "Concepts and Procedures and Logical",
        targetHash: 100,
        targetId: "A",
        targetDescription:
          "Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia",
        targetShortName: "",
        interactionTypeCode: "EQ",
        interactionTypeLabel: "Equation Multiple Choice choice",
        isPerformanceItem: true,
        brailleOnlyItem: false,
        calculator: true
      }
    ],
    [
      {
        bankKey: 187,
        itemKey: 3615,
        title: "Math Grade 6 Claim 4",
        grade: 8,
        gradeLabel: "Grade 6",
        subjectCode: "MATH",
        subjectLabel: "Math",
        claimCode: "MATH4",
        claimLabel: "Modeling/Data Analysis",
        targetHash: 0,
        targetId: "C",
        targetDescription: "target C description",
        targetShortName: "",
        interactionTypeCode: "MS",
        interactionTypeLabel: "Multi Select",
        isPerformanceItem: true,
        brailleOnlyItem: false
      }
    ]
  ]
};
