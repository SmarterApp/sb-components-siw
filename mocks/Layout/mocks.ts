import { SbNavlinkProps } from "@src/index";

export const mockSiteLinks: SbNavlinkProps[] = [
  { name: "Home", url: "url1" },
  { name: "About Test Items", url: "url2" },
  { name: "Browse Items", url: "url3" }
];
