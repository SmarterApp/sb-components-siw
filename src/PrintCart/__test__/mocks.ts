import { ItemCardModel } from "@src/index";
import { currentId } from "async_hooks";

export const itemHandler = jest.fn((item: ItemCardModel) => {
  return undefined;
});

export const getSelectedItemCount = jest.fn(() => {
  return;
});

export const countNumberOfItemsAfterSelection = jest.fn(
  (currentItems: ItemCardModel[], selectedItemsCount: number) => {
    return;
  }
);

export const onAddOrRemoveSelectedItems = jest.fn((item: ItemCardModel) => {
  return;
});

export const onItemsReorder = jest.fn((i: number, j: number) => {
  return;
});
