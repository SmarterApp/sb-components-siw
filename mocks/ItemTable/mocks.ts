import {
  ItemTableProps,
  ItemTableContainerProps,
  ItemCardModel
} from "@src/index";
import { aboutItemMockModel } from "@mocks/AboutItem/mocks";
import { itemCardList, sortableItemCards } from "@mocks/index";

export const tabClassNames = [
  "item",
  "claim",
  "standard",
  "subject",
  "grade",
  "item-type",
  "Target"
];

export const itemTableProps: ItemTableContainerProps = {
  isInterimSite: false,
  isLinkTable: false,
  onRowSelection: () => {
    return;
  },
  onItemSelection: () => {
    return;
  },
  onCountNumberOfItemSelection: () => {
    return;
  },
  itemCards: itemCardList,
  item: {
    kind: "none"
  },

  numberOfSelectedItem: 0,
  getSelectedItemCount: () => {
    return 0;
  },
  showErrorModalOnPrintItemsCountExceeded: () => {
    return;
  },
  associatedItems: [],
  countNumberOfItemsAfterSelection: () => {
    return 0;
  },
  testCodeToLabelMap: { "Test-Name-1": "Test 1", "Test-Name-2": "Test 2" }
};

export const itemTableSortProps: ItemTableContainerProps = {
  ...itemTableProps,
  item: {
    kind: "none"
  },
  itemCards: sortableItemCards
};

export const itemTableSelectProps: ItemTableContainerProps = {
  ...itemTableProps,
  item: {
    kind: "success",
    content: aboutItemMockModel
  }
};
