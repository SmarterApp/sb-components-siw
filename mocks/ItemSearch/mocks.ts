import { ItemCardModel } from "@src/ItemCard/ItemCardModels";
import {
  SearchFilterStringTypes,
  TargetModel,
  SearchAPIParamsModel,
  ItemsSearchModel,
  FilterCategoryModel,
  FilterOptionModel,
  FilterType,
  GradeLevels,
  GradeLevel
} from "@src/index";
import { ClaimModel } from "@src/index";

export const genericSearchStringTypes: SearchFilterStringTypes[] = [
  { label: "test", code: "t1" },
  { label: "test2", code: "t2" },
  { label: "test3", code: "t3" }
];

export const claimsMocks: ClaimModel[] = [
  { label: "test", code: "t1", claimNumber: "1" },
  { label: "test2", code: "t2", claimNumber: "2" },
  { label: "test3", code: "t3", claimNumber: "3" }
];

export const resultFilterOptionModel: FilterOptionModel[] = [
  {
    filterType: FilterType.Subject,
    isSelected: false,
    key: "t1",
    label: "test"
  },
  {
    filterType: FilterType.Subject,
    isSelected: false,
    key: "t2",
    label: "test2"
  },
  {
    filterType: FilterType.Subject,
    isSelected: false,
    key: "t3",
    label: "test3"
  }
];

export const resultFilterOptionModelSelected: FilterOptionModel[] = [
  {
    filterType: FilterType.Subject,
    isSelected: true,
    key: "t1",
    label: "test"
  },
  {
    filterType: FilterType.Subject,
    isSelected: false,
    key: "t2",
    label: "test2"
  },
  {
    filterType: FilterType.Subject,
    isSelected: true,
    key: "t3",
    label: "test3"
  }
];

export const resultFilterGradeModel: FilterOptionModel[] = [
  {
    filterType: FilterType.Grade,
    isSelected: false,
    key: "1",
    label: "Grade 3"
  },
  {
    filterType: FilterType.Grade,
    isSelected: false,
    key: "2",
    label: "Grade 4"
  }
];

export const resultFilterGradeModelSelectedSingle: FilterOptionModel[] = [
  {
    filterType: FilterType.Grade,
    isSelected: true,
    key: "1",
    label: "Grade 3"
  },
  {
    filterType: FilterType.Grade,
    isSelected: false,
    key: "2",
    label: "Grade 4"
  }
];

export const resultFilterGradeModelSelectedMultiple: FilterOptionModel[] = [
  {
    filterType: FilterType.Grade,
    isSelected: true,
    key: "1",
    label: "Grade 3"
  },
  {
    filterType: FilterType.Grade,
    isSelected: true,
    key: "2",
    label: "Grade 4"
  }
];

export const searchOptionFilterTarget: TargetModel[] = [
  { idLabel: "A", id: "A-1", name: "steak sauce", nameHash: 1 },
  { idLabel: "B", id: "B-1", name: "test", nameHash: 2 },
  { idLabel: "C", id: "C-4", name: "boom", nameHash: 3 }
];

export const resultSearchOptionFilterTarget: FilterOptionModel[] = [
  { label: "A", key: "A", isSelected: false, filterType: FilterType.Target },
  {
    label: "B",
    key: "B",
    isSelected: false,
    filterType: FilterType.Target
  },
  { label: "C", key: "C", isSelected: false, filterType: FilterType.Target }
];

export const resultSearchOptionFilterTargetSelectedSingle: FilterOptionModel[] = [
  { label: "A", key: "A", isSelected: true, filterType: FilterType.Target },
  {
    label: "B",
    key: "B",
    isSelected: false,
    filterType: FilterType.Target
  },
  { label: "C", key: "C", isSelected: false, filterType: FilterType.Target }
];

export const resultSearchOptionFilterTargetSelectedMultiple: FilterOptionModel[] = [
  { label: "A", key: "A", isSelected: true, filterType: FilterType.Target },
  {
    label: "B",
    key: "B",
    isSelected: false,
    filterType: FilterType.Target
  },
  { label: "C", key: "C", isSelected: true, filterType: FilterType.Target }
];

export const mockSeachAPI: SearchAPIParamsModel = {
  itemId: "item1",
  // tslint:disable-next-line:no-bitwise
  gradeLevels: GradeLevels.Grade3 | GradeLevels.Grade4,
  subjects: ["t1", "t2"],
  testNames: ["Test-Name-1"],
  claims: ["t1", "t2"],
  interactionTypes: ["t1", "t2"],
  performanceOnly: true,
  catOnly: true,
  targets: ["A", "B"]
};

export const resultFilterOptionModelClaim: FilterOptionModel[] = [
  {
    filterType: FilterType.Claim,
    isSelected: true,
    key: "t1",
    label: "test"
  },
  {
    filterType: FilterType.Claim,
    isSelected: true,
    key: "t2",
    label: "test2"
  },
  {
    filterType: FilterType.Claim,
    isSelected: false,
    key: "t3",
    label: "test3"
  }
];

export const resultFilterOptionModelIT: FilterOptionModel[] = [
  {
    filterType: FilterType.InteractionType,
    isSelected: true,
    key: "t1",
    label: "test"
  },
  {
    filterType: FilterType.InteractionType,
    isSelected: true,
    key: "t2",
    label: "test2"
  },
  {
    filterType: FilterType.InteractionType,
    isSelected: false,
    key: "t3",
    label: "test3"
  }
];

export const resultFilterOptionModelSubject: FilterOptionModel[] = [
  {
    filterType: FilterType.Subject,
    isSelected: true,
    key: "t1",
    label: "test"
  },
  {
    filterType: FilterType.Subject,
    isSelected: true,
    key: "t2",
    label: "test2"
  },
  {
    filterType: FilterType.Subject,
    isSelected: false,
    key: "t3",
    label: "test3"
  }
];

export const gradeSearchStringTypes: GradeLevels[] = [
  GradeLevels.Grade3,
  GradeLevels.Grade4,
  GradeLevels.Grade5
];

export const resultFilterOptionModelGrade: FilterOptionModel[] = [
  {
    filterType: FilterType.Grade,
    isSelected: true,
    key: "1",
    label: "Grade 3"
  },
  {
    filterType: FilterType.Grade,
    isSelected: true,
    key: "2",
    label: "Grade 4"
  },
  {
    filterType: FilterType.Grade,
    isSelected: false,
    key: "4",
    label: "Grade 5"
  }
];

export const targetSearchStringTypes: TargetModel[] = [
  { idLabel: "A", id: "A-1", name: "steak sauce", nameHash: 1 },
  { idLabel: "B", id: "B-1", name: "test", nameHash: 2 },
  { idLabel: "C", id: "C-4", name: "boom", nameHash: 3 }
];

export const resultFilterOptionModeltarget: FilterOptionModel[] = [
  {
    filterType: FilterType.Target,
    isSelected: true,
    key: "A",
    label: "A"
  },
  {
    filterType: FilterType.Target,
    isSelected: true,
    key: "B",
    label: "B"
  },
  {
    filterType: FilterType.Target,
    isSelected: false,
    key: "C",
    label: "C"
  }
];

export const techtypeSearchStringTypes: SearchFilterStringTypes[] = [
  { label: "test", code: "CAT" },
  { label: "test2", code: "Performance" },
  { label: "test3", code: "t3" }
];

export const resultFilterOptionModelTechType: FilterOptionModel[] = [
  {
    filterType: FilterType.TechnologyType,
    isSelected: true,
    key: "CAT",
    label: "test"
  },
  {
    filterType: FilterType.TechnologyType,
    isSelected: true,
    key: "Performance",
    label: "test2"
  },
  {
    filterType: FilterType.TechnologyType,
    isSelected: false,
    key: "t3",
    label: "test3"
  }
];

export const itemSearchModel: ItemsSearchModel = {
  subjects: [
    {
      label: "Mathematics",
      code: "MATH",
      claimCodes: ["MATH1", "MATH2"],
      interactionTypeCodes: ["IT1", "IT2", "IT3"]
    },
    {
      label: "English",
      code: "ELA",
      claimCodes: ["ELA1", "ELA2"],
      interactionTypeCodes: ["IT3", "IT4", "IT5"]
    }
  ],
  claims: [
    {
      label: "Math 1",
      code: "MATH1",
      targetCodes: ["1", "2", "3", "4"],
      claimNumber: "1"
    },
    {
      label: "Math 2",
      code: "MATH2",
      targetCodes: ["5", "2", "3", "4"],
      claimNumber: "2"
    },
    {
      label: "Math 3",
      code: "MATH3",
      targetCodes: ["6", "2", "3", "4"],
      claimNumber: "3"
    },
    {
      label: "English 1",
      code: "ELA1",
      targetCodes: ["5", "6", "7", "8"],
      claimNumber: "1"
    },
    {
      label: "English 2",
      code: "ELA2",
      targetCodes: ["8", "9", "10", "11"],
      claimNumber: "2"
    }
  ]
};
