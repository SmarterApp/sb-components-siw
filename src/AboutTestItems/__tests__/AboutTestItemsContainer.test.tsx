import * as React from "react";
import * as ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";
import * as TestUtils from "react-dom/test-utils";
import { MemoryRouter } from "react-router-dom";
import {
  aboutTestMatch,
  aboutTestPath,
  aboutTestBadItem,
  defaultAboutTestItemsModel
} from "@mocks/AboutTestItems/mocks";
import {
  AboutItem,
  ItemViewerFrame,
  Select,
  AboutTestItemsContainer
} from "@src/index";
import * as Mocks from "./mocks";

describe("AboutTestItemsContainer", () => {
  const wrapper = shallow(
    <AboutTestItemsContainer
      showRubrics={true}
      aboutClient={Mocks.mockedAboutClient}
      params={{}}
      errorRedirectPath=""
    />
  );

  it("default render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("about item modal", () => {
    const aboutItem = wrapper.findWhere(node => node.type() === AboutItem);
    expect(aboutItem).toBeDefined();
    expect(aboutItem).toMatchSnapshot();
  });
  it("item frame", () => {
    const itemFrame = wrapper.findWhere(
      node => node.type() === ItemViewerFrame
    );
    expect(itemFrame).toBeDefined();
    expect(itemFrame).toMatchSnapshot();
  });

  it("sets about item", () => {
    const wrapperInstance = wrapper.instance() as AboutTestItemsContainer;
    wrapperInstance.onFetchedUpdatedViewModel(defaultAboutTestItemsModel);
    const {
      aboutItemsViewModel,
      aboutThisItemViewModel
    } = wrapperInstance.state;
    expect(aboutItemsViewModel.kind).toEqual("success");
    expect(aboutThisItemViewModel.kind).toEqual("success");
  });

  it("changes selected code", () => {
    const wrapperInstance = wrapper.instance() as AboutTestItemsContainer;
    wrapperInstance.handleChange("WER");
    expect(wrapperInstance.state.selectedCode).toEqual("WER");
  });

  it("calls about client", () => {
    const wrapperInstance = wrapper.instance() as AboutTestItemsContainer;
    wrapperInstance.fetchUpdatedViewModel("WER");
    expect(Mocks.mockedAboutClient).toHaveBeenCalled();
  });
  it("sets selected code", () => {
    const wrapperInstance = wrapper.instance() as AboutTestItemsContainer;
    wrapperInstance.fetchUpdatedViewModel("WER");
    expect(wrapperInstance.state.selectedCode).toEqual("WER");
  });
});
