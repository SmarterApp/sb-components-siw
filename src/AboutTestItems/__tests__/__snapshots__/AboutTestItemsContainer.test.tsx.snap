// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutTestItemsContainer about item modal 1`] = `null`;

exports[`AboutTestItemsContainer default render 1`] = `
<LoadingOverlay
  loading={true}
>
  <div
    className="container about-items"
  >
    <div
      className="about-items-info"
    >
      <h2
        className="page-title"
      >
        About Test Items
      </h2>
      <div
        className="section section-light"
      >
        <p
          className="about-items-text"
        >
          Smarter Balanced assessments use a variety of item types to accurately measure what students know and can do. To learn more and see an example item, select an item type below.
        </p>
      </div>
    </div>
    <div
      className="section about-items-iframe"
    >
      <div
        className="section section-light no-item itemViewerFrame"
      >
        <p>
          Please Select an Item Type
        </p>
      </div>
    </div>
  </div>
</LoadingOverlay>
`;

exports[`AboutTestItemsContainer item frame 1`] = `null`;
